packages/openapi-validator/lib/utils/OpenApi3Spec.utils.js

```diff
@@ -0,0 +1,91 @@
+const generateCombinations = require('combos');
+const { defaultBasePath } = require('./common.utils');
+
+const unique = (array) => [...new Set(array)];
+
+const serversPropertyNotProvidedOrIsEmptyArray = (spec) =>
+  !Object.prototype.hasOwnProperty.call(spec, 'servers') ||
+  !spec.servers.length;
+
+const getBasePath = (url) => {
+  const basePathStartIndex = url.replace('//', '  ').indexOf('/');
+  return basePathStartIndex !== -1
+    ? url.slice(basePathStartIndex)
+    : defaultBasePath;
+};
+
+const getPossibleValuesOfServerVariable = ({
+  default: defaultValue,
+  enum: enumMembers,
+}) =>
+  enumMembers ? unique([defaultValue].concat(enumMembers)) : [defaultValue];
+
+const mapServerVariablesToPossibleValues = (serverVariables) =>
+  Object.entries(serverVariables).reduce(
+    (currentMap, [variableName, detailsOfPossibleValues]) => ({
+      ...currentMap,
+      [variableName]: getPossibleValuesOfServerVariable(
```

Comment:
note: we use `variableName` here but `variable` on line 39. I think by definition a variable _is_ a variableName, but in the openapi spec a server `variable` is a complex object. This is an example of how migrating our code to TypeScript in future will hopefully help clarify it